python/lsst/ap/verify/ingestion.py

```diff
@@ -0,0 +1,480 @@
+#
+# LSST Data Management System
+# Copyright 2017 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+"""Data ingestion for ap_verify.
+
+This module handles ingestion of a dataset into an appropriate repository, so
+that pipeline code need not be aware of the dataset framework.
+
+At present, the code is specific to DECam; it will be generalized to other
+instruments in the future.
+"""
+
+from __future__ import absolute_import, division, print_function
+
+__all__ = ["ingestDataset"]
+
+import os
+import tarfile
+from glob import glob
+import sqlite3
+
+import lsst.log
+from lsst.obs.decam import ingest
+from lsst.obs.decam import ingestCalibs
+from lsst.obs.decam.ingest import DecamParseTask
+from lsst.pipe.tasks.ingest import IngestConfig
+from lsst.pipe.tasks.ingestCalibs import IngestCalibsConfig, IngestCalibsTask
+from lsst.pipe.tasks.ingestCalibs import IngestCalibsArgumentParser
+import lsst.daf.base as dafBase
+from lsst.ap.pipe import doIngestTemplates
+
+# Names of directories to be created in specified repository
+_INGESTED_DIR = 'ingested'
+_CALIBINGESTED_DIR = 'calibingested'
+
+# Name of defects tarball residing in dataset's defects directory
+_DEFECT_TARBALL = 'defects_2014-12-05.tar.gz'
+
+
+def ingestDataset(dataset, repository):
+    """Ingest the contents of a dataset into a Butler repository.
+
+    The original data directory shall not be modified.
+
+    Parameters
+    ----------
+    dataset : `lsst.ap.verify.dataset.Dataset`
+        The dataset to be ingested.
+    repository : `str`
+        The file location of a repository to which the data will be ingested.
+        Shall be created if it does not exist. If ``repository`` does exist, it
+        must be compatible with ``dataset`` (in particular, it must support the
+        correct instrument).
```

Comment:
Not immediately clear what is meant by "it must support the correct instrument."

Reply:
I mean the way the `_mapper` needs to know about the obs package it's supposed to be using. Is there an implementation-independent way of saying that?

Reply:
Ah, I see. Perhaps just say "it must support the relevant obs_ package" ?